executor/src/lib.rs

```diff
@@ -620,12 +624,18 @@ where
                 txn_data.gas_used(),
                 txn_data.status().vm_status().major_status,
             ));
+            let reconfiguration_key = ValidatorSet::change_event_key();
+            for event in txn_data.events() {
+                if *event.key() == reconfiguration_key {
+                    reconfiguration_events.push(event.clone());
```

Comment:
this doesn't seem consistent with above(which returns all events).
I thought we'll have a subscribe_to_event_key(event_key) so suggested we filter out events in state sync, but we can start with subscribe() just for Reconfig event. 
either way just make these two functions consistent

Reply:
For full nodes I was treating `executor` as part of state sync - even if we move out the filtering logic of `execute_and_commit_chunk`, filtering through all events and publishing the result to subscribed components will still happen in `execute_chunk` (also in executor). Are you okay with this? cc @phoenix-antigravity

Reply:
talked offline - for now we will keep the event filtering logic inside executor for both full node and validator.